### ISSUE TITLE
SA1210 Ordering is incorrect when having same namespace lower case and capital

### ISSUE THREAD

**Issue by gps7** [DotNetAnalyzers/StyleCopAnalyzers#2971]
Rule validation is incorrect usings when having same namespace but with different case:
using Test.Sample.Model
using Test.SAMPLE.Equipment

The order above is correct. The StyleCop classic works correctly but not Roslyn based.

**Comment by vyrp** [comment]
@gps7: I think the order in your question is incorrect. I don't know if StyleCopAnalyzers is case sensitive or not, but both case-sensitive and case-insensitive comparisons say that "Test.Sample.Model" is bigger than "Test.SAMPLE.Equipment", and therefore should come **after** "Test.SAMPLE.Equipment":
```csharp
Console.WriteLine(string.Compare("Test.Sample.Model", "Test.SAMPLE.Equipment", StringComparison.Ordinal) > 0); // True
Console.WriteLine(string.Compare("Test.Sample.Model", "Test.SAMPLE.Equipment", StringComparison.OrdinalIgnoreCase) > 0); // True
```

**Comment by gps7** [comment]
@vyrp : The testing this with C# or Excel always gives this results. So I did further investigation and used the  simple comparison of same string combinations:

Unsorted: | After   sorting:
-- | --
ABC | abc
aBc | aBc
aBC | aBC
Abc | Abc
ABC | ABc
ABc | ABC
abc | ABC

However, after adding numbers as Namespace resulted in no change:

Unsorted: | After   sorting:
-- | --
ABC.1 | ABC.1
aBc.2 | aBc.2
aBC.3 | aBC.3
Abc.4 | Abc.4
ABC.5 | ABC.5
ABc.6 | ABc.6
abc.7 | abc.7

This looks like the comparison of namespace is incorrect since it should be divided before comparing. Isn't it?

Basically it is down to results based on each namespace comparison order:
aaa.A
aaa.W
AAA.A
AAA.W

or based on full namespace comparison results in different order:
aaa.A
AAA.A
aaa.W
AAA.W

**Comment by vyrp** [comment]
@gps7: I'm sorry, my mistake. I used Ordinal comparison instead of InvariantCulture. With InvariantCulture, indeed "Sample" comes before "SAMPLE".

I looked at [the code](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/29cd20779e8e354b68f138710cd263fdf5e72f69/StyleCop.Analyzers/StyleCop.Analyzers/Helpers/NameSyntaxHelpers.cs#L54-L59) and it's comparing the full namespace string, not its parts. And it's doing so ignoring case. In case of tie, it then compares considering case.

However, the outcome of the logic above is that the correct order deemed by StyleCop.Analyzers is:
```csharp
using Sample;
using SAMPLE;
using SAMPLE.Equipment;
using Sample.Model;
```
Which indeed looks inconsistent. I wonder if that is a bug or it is expected. I will let the repo owners comment on that.

**Comment by vweijsters** [comment]
Both the original StyleCop and VS2019 suggest that 
```c#
using Test.Sample.Model;
using Test.SAMPLE.Equipment;
```
is the correct order, so we may as well change our implementation to match that.
@sharwell : What are your thoughts on this?

**Comment by sharwell** [comment]
I'm still trying to convince myself. I lean towards changing the approach to use a natural order, where later word parts are only compared if earlier parts are identical:

* `abc2` comes before `ABC10`
* `abc10` comes before `ABC2`
* `ab.d` comes before `Ab.c`

Alias qualifiers would [continue to be treated](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/KnownChanges.md#sa1210) as segments of the using directive.

**Comment by webber2k6** [comment]
Hi guys, I'm currently facing the same issue with namespaces from external libraries.
Visiual Studio sorts the usings case sensitive but StyleCop is complaining about this order:

* Hr.Smart.*
* HR.Sirius.*

I'm using the most recent unstable version: 1.2.0.435
Are there any plans to fix this issue?
